packages/agoric-cli/src/lib/psm.js

```diff
@@ -0,0 +1,123 @@
+// @ts-check
+
+import { COSMOS_UNIT, makeAmountFormatter } from './format.js';
+// eslint-disable-next-line no-unused-vars -- typeof below
+import { makeAgoricNames } from './rpc.js';
+
+// Ambient types. Needed only for dev but this does a runtime import.
+import '@agoric/zoe/src/zoeService/types.js';
+
+/** @typedef {import('@agoric/smart-wallet/src/offers').OfferSpec} OfferSpec */
+/** @typedef {import('@agoric/smart-wallet/src/offers').OfferStatus} OfferStatus */
+/** @typedef {import('@agoric/smart-wallet/src/smartWallet').BridgeAction} BridgeAction */
+
+/**
+ * @param {{ assets: import('./format').AssetDescriptor[], offers: Map<number, OfferStatus>}} state
+ * @param {Awaited<ReturnType<typeof makeAgoricNames>>} agoricNames
+ */
+export const simpleOffers = (state, agoricNames) => {
+  const { assets, offers } = state;
+  const fmt = makeAmountFormatter(assets);
+  const fmtRecord = r =>
+    Object.fromEntries(
+      Object.entries(r).map(([kw, amount]) => [kw, fmt(amount)]),
+    );
+  return [...offers.keys()].sort().map(id => {
+    const o = offers.get(id);
+    assert(o);
+    assert(o.invitationSpec.source === 'contract');
+    const {
+      invitationSpec: { instance, publicInvitationMaker },
+      proposal: { give, want },
+      payouts,
+    } = o;
+    const entry = Object.entries(agoricNames.instance).find(
+      ([_name, candidate]) => candidate === instance,
+    );
+    const instanceName = entry ? entry[0] : '???';
+    // console.log({ give: JSON.stringify(give), want: JSON.stringify(want) });
+    return [
+      instanceName,
+      new Date(id).toISOString(),
+      id,
+      publicInvitationMaker,
+      o.numWantsSatisfied,
+      {
+        give: fmtRecord(give),
+        want: fmtRecord(want),
+        ...(payouts ? { payouts: fmtRecord(payouts) } : {}),
+      },
+    ];
+  });
+};
+
+/**
+ * @param {Record<string, Brand>} brands
+ * @param {({ wantMinted: number | undefined, wantAnchor: number | undefined })} opts
+ * @param {number} [fee=0]
+ * @param {string} [anchor]
+ * @returns {Proposal}
+ */
+export const makePSMProposal = (brands, opts, fee = 0, anchor = 'AUSD') => {
+  const direction = opts.wantAnchor ? 'giveMinted' : 'giveAnchor';
+  const brand =
+    direction === 'giveAnchor'
+      ? { in: brands[anchor], out: brands.IST }
+      : { in: brands.IST, out: brands[anchor] };
+  const value =
+    Number(direction === 'giveAnchor' ? opts.wantMinted : opts.wantAnchor) *
+    Number(COSMOS_UNIT);
+  const adjusted = {
+    in: BigInt(
+      Math.ceil(direction === 'giveAnchor' ? value / (1 - fee) : value),
+    ),
+    out: BigInt(
+      Math.ceil(direction !== 'giveAnchor' ? value * (1 - fee) : value),
+    ),
+  };
```

Comment:
This doesn't look right. The change from `wantMinted` to `giveAnchor` should have resulted in a change in these fee calculations, but they look the same as
https://gist.github.com/dckc/8b5b2f16395cb4d7f2ff340e0bc6b610#file-psm-tool-L70-L73

nearly critical... it should be the case that if they use exactly the fees given in the contract, their trade will work as expected.

But I suppose the user can tweak the fee options to make up the difference, for now.